### PR TITLE
fix: copy to build folder and use for Homey Cloud

### PR DESCRIPTION
**Pull request by tjallingt**
Instead of rewriting file paths when tar-ing this PR copies all app files over to a `build/` directory.
We can then tar this folder for Homey Pro or mount it in the container for Homey Cloud.

This appears to perform reasonably for `com.fibaro` which I think is considered a large app. If performance becomes an issue we could try to avoid copying `node_modules`. I _think_ we can mount it separately inside the docker container and append it to the tar.

Since there where concerns about the performance of copying the files to a folder I measured and optimized the `preprocess()` step:
- `npm prune` turned out to be pretty slow, we can use `npm ls` instead which is faster but gives basically the same result.
- `ignore-walk` doesn't scale well with the number of rules, so instead of ignoring dev dependencies while walking we can ignore `node_modules` and collect the production dependencies separately.

### CODE REVIEW

File: lib/App.js [hunk 1]
```diff
@@ -1107,17 +1079,29 @@ class App {
...
-    walker.onReadIgnoreFile('', ignoreRules.join('\r\n'), () => {});
+    const prunePaths = await NpmCommands.getPrunePaths({ path: this.path });
+
+    const ignoreRules = [
+      '.*',
+      '*.ts',
+      'tsconfig.json',
+      'env.json',
+      '*.compose.json',
+      'build/*',
```
Comment (tjallingt) [inline lib/App.js:1100]: `*.compose.json` and `build/*` where added to the ignore rules. We run compose before copying the "source" files so we don't need the compose files in the `build/` anymore. We obviously don't want to copy `build/` into `build/`

**Comment by tjallingt** [comment]
Included some cleanup commits for the Npm Commands.

**Review by mrlukasbos — APPROVED** [review]
These are quite a bit of scary changes. 

I have tested it and it seems to work fine however. 
I have tested it with a javascript application, converted it to typescript (adding a tsconfig and using the add-types" command) and tested it again. In both tests I ran the app, made it crash once, ran validations and even published.

**Comment by tjallingt** [comment]
Nice thanks! With this many scary changes i'm planning on testing this extremely thoroughly before publishing. But its good to hear its also working on other machines ;P

**Pull request merged**